h2o-py/tests/testdir_algos/gbm/pyunit_effective_parameters_gbm.py

```diff
@@ -0,0 +1,104 @@
+    assert gbm1.parms['categorical_encoding']['actual_value'] == gbm2.parms['categorical_encoding']['actual_value']
+
+    gbm1 = H2OGradientBoostingEstimator(seed = 1234, nfolds=5)
+    gbm1.train(x=predictors, y=response, training_frame=train, validation_frame=valid)
+
+    gbm2 = H2OGradientBoostingEstimator(seed = 1234, nfolds=5, fold_assignment='Random', distribution="bernoulli",
+                                        histogram_type="UniformAdaptive", categorical_encoding="Enum")
+    gbm2.train(x=predictors, y=response, training_frame=train, validation_frame=valid)
+
+    assert gbm1.logloss() == gbm2.logloss()
+    assert gbm1.parms['distribution']['input_value'] == 'AUTO'
+    assert gbm1.parms['distribution']['actual_value'] == gbm2.parms['distribution']['actual_value']
+    assert gbm1.parms['stopping_metric']['input_value'] == 'AUTO'
+    assert gbm1.parms['stopping_metric']['actual_value'] is None
+    assert gbm1.parms['histogram_type']['input_value'] == 'AUTO'
+    assert gbm1.parms['histogram_type']['actual_value'] == gbm2.parms['histogram_type']['actual_value']
+    assert gbm1.parms['fold_assignment']['input_value'] == 'AUTO'
+    assert gbm1.parms['fold_assignment']['actual_value'] == gbm2.parms['fold_assignment']['actual_value']
+    assert gbm1.parms['categorical_encoding']['input_value'] == 'AUTO'
+    assert gbm1.parms['categorical_encoding']['actual_value'] == gbm2.parms['categorical_encoding']['actual_value']
+
+    h2o.rapids("(setproperty \"{}\" \"{}\")".format("sys.ai.h2o.algos.evaluate_auto_model_parameters", "false"))
+
+    gbm1 = H2OGradientBoostingEstimator(seed = 1234, nfolds=5)
+    gbm1.train(x=predictors, y=response, training_frame=train, validation_frame=valid)
+
+    gbm2 = H2OGradientBoostingEstimator(seed = 1234, nfolds=5, fold_assignment='Random', distribution="bernoulli",
+                                    histogram_type="UniformAdaptive", categorical_encoding="Enum")
+    gbm2.train(x=predictors, y=response, training_frame=train, validation_frame=valid)
+
+    assert gbm1.logloss() == gbm2.logloss()
+    assert gbm1.parms['distribution']['input_value'] == 'AUTO'
+    # distribution value was being set before
+    assert gbm1.parms['distribution']['actual_value'] == gbm2.parms['distribution']['actual_value']
+    assert gbm1.parms['stopping_metric']['input_value'] == 'AUTO'
+    assert gbm1.parms['stopping_metric']['actual_value'] == 'AUTO'
+    assert gbm1.parms['histogram_type']['input_value'] == 'AUTO'
+    assert gbm1.parms['histogram_type']['actual_value'] == 'AUTO'
+    assert gbm1.parms['fold_assignment']['input_value'] == 'AUTO'
+    assert gbm1.parms['fold_assignment']['actual_value'] == 'AUTO'
+    assert gbm1.parms['categorical_encoding']['input_value'] == 'AUTO'
+    assert gbm1.parms['categorical_encoding']['actual_value'] == 'AUTO'
+
+    h2o.rapids("(setproperty \"{}\" \"{}\")".format("sys.ai.h2o.algos.evaluate_auto_model_parameters", "true"))
```

Comment:
I think this should be in a finally block since it might break other tests otherwise